### ISSUE TITLE
Shadow variable definition

### ISSUE THREAD

**Issue by Alex97** [NationalSecurityAgency/ghidra#1612]
**Is your feature request related to a problem? Please describe.**
I have come across multiple situations where the memory location of a variable or its associated
register is reused after the variable goes out of scope. For example, if we have a function that has
2 variables, **a** and **b**, both integers, with **b** being used only after **a** goes out of scope,
the compiler can store the value of **b** at the same memory location as **a**. The same principle
applies if the variables are kept in registers.

**Describe the solution you'd like**
It would be helpful if subsequent assignments to a variable could shadow previous usage.
i.e. replace the label of **a** with **b** from a certain point onward, while still referring to the same
memory location.

**Comment by astrelsky** [comment]
Related to/Duplicate of #1510 .

**Comment by saruman9** [comment]
> Related to/Duplicate of #1510.

Also #975.

**Comment by Alex97** [comment]
Ok, I will close this one and continue there, thank you.
